File: app/src/main/java/me/ykrank/s1next/view/internal/LoadingViewModelBindingDelegateBaseCardViewContainerImpl.java
package me.ykrank.s1next.view.internal;

import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.github.ykrank.androidtools.ui.internal.LoadingViewModelBindingDelegate;
import com.github.ykrank.androidtools.ui.vm.LoadingViewModel;

import me.ykrank.s1next.databinding.FragmentBaseCardViewContainerBinding;

public final class LoadingViewModelBindingDelegateBaseCardViewContainerImpl
        implements LoadingViewModelBindingDelegate {

    private final FragmentBaseCardViewContainerBinding binding;

    public LoadingViewModelBindingDelegateBaseCardViewContainerImpl(
            FragmentBaseCardViewContainerBinding binding) {
        this.binding = binding;
    }

    @Override
    public View getRootView() {
        return binding.getRoot();
    }

    @Override
    public SwipeRefreshLayout getSwipeRefreshLayout() {
        return binding.swipeRefreshLayout;
    }

    @Override
    public RecyclerView getRecyclerView() {
        return binding.recyclerView;
    }

    @Override
    public void setLoadingViewModel(LoadingViewModel loadingViewModel) {
        binding.setLoadingViewModel(loadingViewModel);
    }
}

File: app/src/main/java/me/ykrank/s1next/view/internal/LoadingViewModelBindingDelegateBaseCardViewContainerImpl.kt
package me.ykrank.s1next.view.internal

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.github.ykrank.androidtools.ui.internal.LoadingViewModelBindingDelegate
import com.github.ykrank.androidtools.ui.vm.LoadingViewModel
import me.ykrank.s1next.databinding.FragmentBaseCardViewContainerBinding

class LoadingViewModelBindingDelegateBaseCardViewContainerImpl(
    private val binding: FragmentBaseCardViewContainerBinding
) : LoadingViewModelBindingDelegate {
    override val rootView: View
        get() = binding.root
    override val swipeRefreshLayout: SwipeRefreshLayout
        get() = binding.swipeRefreshLayout
    override val recyclerView: RecyclerView
        get() = binding.recyclerView

    override val hintView: TextView
        get() = binding.tvHint

    override fun setLoadingViewModel(loadingViewModel: LoadingViewModel) {
        binding.setLoadingViewModel(loadingViewModel)
    }
}
